pkg/config/config_test.go

```diff
@@ -719,6 +719,36 @@ func TestLoadProxyEmptyValuePrecedence(t *testing.T) {
 	os.Unsetenv("DD_PROXY_NO_PROXY")
 }
 
+func TestLoadProxyWithoutNoProxy(t *testing.T) {
+	config := setupConf()
+
+	// Don't include cloud metadata URL's in no_proxy
+	config.Set("use_proxy_for_cloud_metadata", true)
+
+	// circleCI set some proxy setting
+	ciValue := os.Getenv("NO_PROXY")
+	os.Unsetenv("NO_PROXY")
+	defer os.Setenv("NO_PROXY", ciValue)
+
+	os.Setenv("DD_PROXY_HTTP", "http_url")
+	os.Setenv("DD_PROXY_HTTPS", "https_url")
```

Comment:
We might want to reset the previous value like for `NO_PROXY` instead of `Unsetenv`

Reply:
Done